web/gds-user-ui/src/components/TrixoQuestionnaireForm/__tests__/TrixoQuestionnaire.spec.tsx

```diff
@@ -0,0 +1,138 @@
+  beforeAll(() => {
+    act(() => {
+      dynamicActivate('en');
+    });
+  });
+
+  it('should render', () => {
+    const { container } = renderComponent();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render the form', () => {
+    const { getByTestId } = renderComponent();
+    const trixoQuestionnaire = getByTestId('trixo-form');
+    expect(trixoQuestionnaire).toBeInTheDocument();
+  });
+
+  it('should render the primary national jurisdiction', () => {
+    const { getByText } = renderComponent();
+    const primaryNationalJurisdiction = getByText('Primary National Jurisdiction');
+    expect(primaryNationalJurisdiction).toBeInTheDocument();
+  });
+
+  it('should render the primary national jurisdiction', () => {
+    const { getByText } = renderComponent();
+    const primaryNationalJurisdiction = getByText('Primary National Jurisdiction');
+    expect(primaryNationalJurisdiction).toBeInTheDocument();
+  });
+
+  it('should render add jurisdiction button', () => {
+    const { getByText } = renderComponent();
+    const addJurisdiction = getByText('Add Jurisdiction');
+    expect(addJurisdiction).toBeInTheDocument();
+  });
+
+  it('should add a new fields when clicking on add jurisdiction button', async () => {
+    const { getByTestId } = renderComponent();
+    const addJurisdiction = getByTestId('trixo-add-jurisdictions-btn');
+    userEvent.click(addJurisdiction);
+    await waitFor(() => {
+      expect(getByTestId('trixo-country')).toBeInTheDocument();
+      expect(getByTestId('trixo-regulator-name')).toBeInTheDocument();
+      expect(getByTestId('trixo-add-jurisdictions-btn')).toBeInTheDocument();
+    });
+  });
+
+  it('should render the required financial transfers question', () => {
+    // check if the input after the data-testid is rendered
+    const { getByTestId } = renderComponent();
+    const wrapper = getByTestId('is-required-financial-transfers');
+    const select = within(wrapper).getByRole('combobox');
+    expect(select).toBeInTheDocument();
+  });
+
+  it('should render the required regulatory program questionnaire', () => {
+    const { getByTestId } = renderComponent();
+    const wrapper = getByTestId('trixo-rule-policies');
+    const select = within(wrapper).getByRole('combobox');
+    expect(select).toBeInTheDocument();
+  });
+
+  it('should render Conducts KYC radio button', () => {
+    const { getByTestId, getByText } = renderComponent();
+
+    const questionnaire = getByText('Conducts KYC before virtual asset transfers');
+
+    expect(questionnaire).toBeInTheDocument();
+
+    const wrapper = getByTestId('trixo-kyc-before-virtual-asset-transfers');
+    const radioButtons = within(wrapper).getByRole('checkbox');
+    expect(radioButtons).toBeInTheDocument();
+    // if we click on the radio button, it should be checked
+    userEvent.click(radioButtons);
+    expect(radioButtons).toBeChecked();
+    // if we click again, it should be unchecked
```

Comment:
I appreciate the comments because it can be easy to not notice `not` on line 100.